### PR TITLE
Add Azure Kubernetes Service Approval Accelerator

### PR DESCRIPTION
**Pull request by TLATER**
This continues the work from #74, addressing the first point of #67.

Updated todo list after the first review (see https://github.com/finos/cloud-service-certification/pull/90#issuecomment-773453487):

- Comments from @alfredtommy:
  - [x] Add info on setting up private clusters
  - [x] In-line documentation references

- Comments from @peterrhysthomas:
  - [x] Improve the section on authorization best practices
  - [x] Add detail on key management
  - [x] Remove detail about "continuous maintenance" topics (such as WAF)
  - [ ] VPC configuration

### CODE REVIEW

**Comment by mcleo-d** [comment]
Hi @TLATER

Thanks for raising this new PR 👍  Can you confirm this supersedes #72 which can be closed in favour of this one?

I have added @peterrhysthomas and @abdullahgarcia as reviewers, alongside @alfredtommy who is mentioned in this comment as the person who gave @tmewett the original feedback.

The PR is now added to the CSC kanban board as `in progress`.

Thank you for your time and effort 👏  💯 💯 💯

**Comment by TLATER** [comment]
@mcleo-d yep, #72 can be closed, those commits are in here too :)

**Comment by mcleo-d** [comment]
Done - https://github.com/finos/cloud-service-certification/pull/72#issuecomment-760130663

Thank you @TLATER 👍

**Comment by mcleo-d** [comment]
This is to confirm the meeting scheduled to review **Add Azure Kubernetes Service Approval Accelerator** #90 with @TLATER and the CSC project prior to accepting the PR.

**Topic:** FINOS CSC - Review : Add Azure Kubernetes Service Approval Accelerator #90
**Time:** Jan 22, 2021 3:00 PM GMT / 10:00AM ET

Join Zoom Meeting
https://zoom.us/j/92304379984

Meeting ID: 923 0437 9984
One tap mobile
+13126266799,,92304379984# US (Chicago)
+13462487799,,92304379984# US (Houston)

Dial by your location
+1 312 626 6799 US (Chicago)
+1 346 248 7799 US (Houston)
+1 646 558 8656 US (New York)
+1 669 900 6833 US (San Jose)
+1 253 215 8782 US (Tacoma)
+1 301 715 8592 US (Washington D.C)
855 880 1246 US Toll-free
877 369 0926 US Toll-free
+1 587 328 1099 Canada
+1 647 374 4685 Canada
+1 647 558 0588 Canada
+1 778 907 2071 Canada
+1 204 272 7920 Canada
+1 438 809 7799 Canada
855 703 8985 Canada Toll-free
Meeting ID: 923 0437 9984
Find your local number: https://zoom.us/u/aTTQ5gMQ9

**Comment by TLATER** [comment]
For reference, my todo list from the review:

- Comments from @alfredtommy:
  - [ ] In-lining documentation references (see https://github.com/finos/cloud-service-certification/pull/90#commitcomment-46244582)

- Comments from @peterrhysthomas:
  - [x] Improving the section on authorization best practices
  - [ ] Adding detail on key management
  - [x] Removing detail about "continuous maintenance" topics (such as WAF)
  - [ ] BPC (?) management
    - In retrospect, I should have checked this note was correct, I'll ask if I can't figure out what I meant

In the future, it might be helpful if we could use the GitHub review features (hint: the "review changes" button in the diff tab), it will automatically build up tasks to comment on/resolve - cuts down on the feedback loop regarding questions like "what was that BPC thing again?" :)

**Comment by TLATER** [comment]
@mcleo-d, @alfredtommy, @peterrhysthomas  - I think that concludes the changes for now, could you have a look to see if they match your expectations? @peterrhysthomas specifically, I think I took a bogus note during the review, any idea what I meant with "BPC"?

**Comment by mcleo-d** [comment]
> @mcleo-d, @alfredtommy, @peterrhysthomas - I think that concludes the changes for now, could you have a look to see if they match your expectations? @peterrhysthomas specifically, I think I took a bogus note during the review, any idea what I meant with "BPC"?

@TLATER - Thanks for chasing. I'll follow up with the team and will get back to you.

**Comment by peterrhysthomas** [comment]
> @mcleo-d, @alfredtommy, @peterrhysthomas - I think that concludes the changes for now, could you have a look to see if they match your expectations? @peterrhysthomas specifically, I think I took a bogus note during the review, any idea what I meant with "BPC"?

I can't remember what BPC meant either, and re-reading the diff didn't trigger anything so happy to leave this for now.

**Review by peterrhysthomas — APPROVED** [review]

**Comment by abdullahgarcia** [comment]
First observation:

The file should exist within a folder called "aks", not "kubernetes". This is to match the convention in the AWS folder. Side note: we need to fix this for the GCP folder too.

File: azure/kubernetes/aks-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
+# Service Approval Accelerator for Azure Kubernetes Service (AKS)
+
+## Identity & access management
```
Comment (abdullahgarcia) [inline azure/kubernetes/aks-kubernetes.md:3]: Identity & Access Management

We need to keep it consistent across the document.

File: azure/kubernetes/aks-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+3.  https://kubernetes.io/docs/reference/access-authn-authz/rbac/
+
+### Privileged Access Management
+
+As discussed, identities should be managed via Azure Active Directory.
+Azure provides guidance and AD functionality for managing privileged identities, such as the Privileged Access Management notification service.
+
+1.  https://docs.microsoft.com/en-us/azure/active-directory/users-groups-roles/directory-admin-roles-secure
+
+## Encryption & secure data management
```
Comment (abdullahgarcia) [inline azure/kubernetes/aks-kubernetes.md:60]: Encryption & Secure Data Management

We need to keep it consistent across the document.

File: azure/kubernetes/aks-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+### Encryption of data at rest
```
Comment (abdullahgarcia) [inline azure/kubernetes/aks-kubernetes.md:62]: Encryption of Data at Rest

We need to keep it consistent across the document.

File: azure/kubernetes/aks-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+There are a variety of methods for authenticating with the Storage platform.
+
+Note that this also applies to fully managed components, such as etcd[3].
+
+1.  https://docs.microsoft.com/en-us/azure/storage/common/storage-introduction
+2.  https://docs.microsoft.com/en-gb/azure/aks/concepts-storage
+3.  https://docs.microsoft.com/en-us/azure/aks/concepts-security#kubernetes-secrets
+4.  https://docs.microsoft.com/en-us/azure/storage/common/storage-service-encryption
+
+### Encryption of data in transit
```
Comment (abdullahgarcia) [inline azure/kubernetes/aks-kubernetes.md:77]: Encryption of Data in Transit

We need to keep it consistent across the document.

File: azure/kubernetes/aks-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+In-transit encryption is generally more an application layer concern and not considered by the lower-layer routing and network services that Kubernetes offers.
+However, it does support "ingress controllers" which can process data at the application level, including decryption of TLS for routing to pods.
+This allows centralised management of TLS at the Kubernetes networking level, even for pod applications which do not support it.
+
+1.  https://docs.microsoft.com/en-gb/azure/aks/concepts-network
+
+### Certificate and key management
```
Comment (abdullahgarcia) [inline azure/kubernetes/aks-kubernetes.md:85]: Certificate and Key Management 

or

Certificate & Key Management

We need to keep it consistent across the document.

File: azure/kubernetes/aks-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+infrastructure) is only available in the Azure Information Protection
+service, which is irrelevant to Kubernetes [2].
+
+1.  https://docs.microsoft.com/en-gb/azure/security/fundamentals/encryption-overview#server-side-encryption
+2.  https://docs.microsoft.com/en-us/azure/information-protection/configure-adrms-restrictions
+3.  https://docs.microsoft.com/en-us/azure/aks/azure-disk-customer-managed-keys
+4.  https://docs.microsoft.com/en-us/azure/key-vault/secrets/overview-storage-keys
+5.  https://github.com/MicrosoftDocs/azure-docs/issues/58185#issuecomment-655014105
+
+## Network security
```
Comment (abdullahgarcia) [inline azure/kubernetes/aks-kubernetes.md:141]: Network Security

We need to keep it consistent across the document.

**Review by abdullahgarcia — COMMENTED** [review]
Hi,

The overall content looks good.

Few minor corrections in terms of format.

Also, I think I did ask about VPC configuration and I believe should be in the section with "Network Isolation" and "IP Firewall Rules". Can we please double check this matter?

Thanks!

**Review by abdullahgarcia — CHANGES_REQUESTED** [review]
Hi,

The overall content looks good.

Few minor corrections in terms of format.

Also, I think I did ask about VPC configuration and I believe should be in the section with "Network Isolation" and "IP Firewall Rules". Can we please double check this matter?

Thanks!

File: azure/kubernetes/aks-kubernetes.md (hunk 1 above)
Comment (mcleo-d) [reply azure/kubernetes/aks-kubernetes.md:3]: @TLATER - I have addressed in the following PR to your fork ...
https://github.com/TLATER/cloud-service-certification/pull/2

File: azure/kubernetes/aks-kubernetes.md (hunk 2 above)
Comment (mcleo-d) [reply azure/kubernetes/aks-kubernetes.md:60]: @TLATER - I have addressed in the following PR to your fork ...

- https://github.com/TLATER/cloud-service-certification/pull/2

File: azure/kubernetes/aks-kubernetes.md (hunk 3 above)
Comment (mcleo-d) [reply azure/kubernetes/aks-kubernetes.md:62]: @TLATER - I have addressed in the following PR to your fork ...

- https://github.com/TLATER/cloud-service-certification/pull/2

File: azure/kubernetes/aks-kubernetes.md (hunk 4 above)
Comment (mcleo-d) [reply azure/kubernetes/aks-kubernetes.md:77]: @TLATER - I have addressed in the following PR to your fork ...

- https://github.com/TLATER/cloud-service-certification/pull/2

File: azure/kubernetes/aks-kubernetes.md (hunk 5 above)
Comment (mcleo-d) [reply azure/kubernetes/aks-kubernetes.md:85]: @TLATER - I have addressed in the following PR to your fork ...

- https://github.com/TLATER/cloud-service-certification/pull/2

File: azure/kubernetes/aks-kubernetes.md (hunk 6 above)
Comment (mcleo-d) [reply azure/kubernetes/aks-kubernetes.md:141]: @TLATER - I have addressed in the following PR to your fork ...

- https://github.com/TLATER/cloud-service-certification/pull/2

**Comment by mcleo-d** [comment]
> First observation:
> 
> The file should exist within a folder called "aks", not "kubernetes". This is to match the convention in the AWS folder. Side note: we need to fix this for the GCP folder too.

@TLATER - I have addressed in the following PR to your fork ...

- https://github.com/finos/cloud-service-certification/pull/90#issuecomment-779155858

**Comment by mcleo-d** [comment]
@TLATER and @abdullahgarcia,

I've gone through the feedback and have raised the following PR against @TLATER's branch where the PR is raised https://github.com/TLATER/cloud-service-certification/pull/2.

I have not addressed the feedback below 👇 but have made comments against all other feedback raised by @abdullahgarcia.

> Also, I think I did ask about VPC configuration and I believe should be in the section with "Network Isolation" and "IP Firewall Rules". Can we please double check this matter?

I hope this approach helps. Please do let me know.

James.

**Comment by TLATER** [comment]
Thanks @mcleo-d - I've merged the commit. Forgive me for re-committing it as well, GitHub's squash-and-merge doesn't quite do to the header what you'd expect it to.

I'll add a note about VPC configuration.

**Review by abdullahgarcia — APPROVED** [review]

**Comment by mcleo-d** [comment]
Congratulation @TLATER, @adds68, @abdullahgarcia and @peterrhysthomas for the work done contributing this pull request. 

Lets celebrate in the next project meeting on 25th February ... #101 😸 

Great team collaboration 👏  💯 💯 💯

**Pull request merged**